improver_tests/psychrometric_calculations/condensation_trails/test_CondensationTrailFormation.py

```diff
@@ -223,3 +223,146 @@ def test_find_local_vapour_pressure(
+                            [210.87348485, 211.51295453, 212.28699057],
+                            [213.28191226, 214.72817613, 219.25],
+                        ],
+                    ],
+                    [
+                        [
+                            [233.95, 234.72353454, 235.71222294],
+                            [236.98378723, 238.83600501, 244.65],
+                        ],
+                        [
+                            [211.96734685, 212.61486738, 213.39860275],
+                            [214.40593674, 215.87057442, 220.45],
+                        ],
+                    ],
+                    [
+                        [
+                            [235.45, 236.26764963, 237.27141769],
+                            [238.56273465, 240.44372252, 246.35],
+                        ],
+                        [
+                            [213.15171974, 213.80810569, 214.60240506],
+                            [215.62327924, 217.10765399, 221.75],
+                        ],
+                    ],
+                ],
+                dtype=np.float32,
+            ),
+            np.array(
+                [
+                    [-1126.49826644, -101.36159382],
+                    [-1274.79858365, -116.00541492],
+                    [-1478.1242258, -134.00868055],
+                ],
+                dtype=np.float32,
+            ),
+        ),
+    ],
+)
+def test_calculate_critical_temperatures_and_intercepts_values(
+    engine_contrail_factors: List[float],
+    pressure_levels: np.ndarray,
+    relative_humidity: np.ndarray,
+    expected_critical_temperatures: np.ndarray,
+    expected_critical_intercepts: np.ndarray,
+):
+    """
+    Test that _calculate_critical_temperatures_and_intercepts() returns two arrays with
+    the expected shapes and values.
+
+    Args:
+        engine_contrail_factors (List[float]) List of contrail factors used to initialise the contrails class (kg/kg/K).
+        pressure_levels (np.ndarray): Array of pressure levels (Pa).
+        relative_humidity (np.ndarray): Array of relative humidity values (kg/kg).
+        expected_critical_temperatures (np.ndarray): Expected critical temperature output on pressure levels. Array axes are [contrail factors, pressure levels, relative humidity] (K).
+        expected_critical_intercepts (np.ndarray): Expected critical intercept output on pressure levels. Array axes are [contrail factors, pressure levels] (Pa).
+    """
+    plugin = CondensationTrailFormation(engine_contrail_factors)
+    plugin.engine_mixing_ratios = plugin._calculate_engine_mixing_ratios(
+        pressure_levels
+    )
+    plugin.relative_humidity = np.broadcast_to(
+        relative_humidity, pressure_levels.shape + relative_humidity.shape
+    )
+    plugin._calculate_critical_temperatures_and_intercepts()
+
+    np.testing.assert_allclose(
+        plugin.critical_temperatures,
+        expected_critical_temperatures,
+        rtol=1e-3,
```

Comment:
The expected numbers have up to eleven significant figures. This says to check three significant figures. Float32 values have a precision of seven s.f. So I suggest changing this to 
```suggestion
        rtol=1e-6,
```
And truncating the expected values to 7 s.f. (one more than we test for, but not eight more).